aws-distro-opentelemetry-node-autoinstrumentation/test/exporter/otlp/aws/common/otlp-aws-base-exporter.test.ts

```diff
@@ -0,0 +1,238 @@
+      },
+      {
+        description: 'Should fail when headers are undefined',
+        test: (done: () => void) => this.testUndefinedHeaders(done),
+      },
+      {
+        description: 'Should continue when authenticate returns undefined',
+        test: (done: () => void) => this.testSigningFails(done),
+      },
+    ];
+  }
+
+  private testSigV4Headers(done: () => void) {
+    const exporterClass = this.getExporter();
+    const exporter = new exporterClass(this.getEndpoint() + this.getEndpointPath());
+
+    exporter
+      .export([], (result: ExportResult) => {
+        expect(result.code).toBe(ExportResultCode.SUCCESS);
+        expect(result.error?.message).toBe(undefined);
+      })
+      .then(() => {
+        this.scope.on('request', (req, interceptor, body) => {
+          this.assertHeaders(req.headers);
+          expect(req.headers).not.toHaveProperty('content-encoding');
+          done();
+        });
+      });
+  }
+
+  private testEnableCompression(done: () => void) {
+    const exporterClass = this.getExporter();
+    const exporter = new exporterClass(this.getEndpoint() + this.getEndpointPath(), {
+      compression: CompressionAlgorithm.GZIP,
+    });
+
+    exporter
+      .export([], (result: ExportResult) => {
+        expect(result.code).toBe(ExportResultCode.SUCCESS);
+        expect(result.error?.message).toBe(undefined);
+      })
+      .then(() => {
+        this.scope.on('request', (req, interceptor, body) => {
+          this.assertHeaders(req.headers);
+          expect(req.headers['content-encoding']).toBe('gzip');
+
+          //Gzip first 10 bytes are reserved for metadata headers:
+          //https://www.loc.gov/preservation/digital/formats/fdd/fdd000599.shtml?loclr=blogsig
+          const data = Buffer.from(body, 'hex');
+          expect(data.length).toBeGreaterThanOrEqual(10);
+          expect(data.subarray(0, 2)).toEqual(Buffer.from([0x1f, 0x8b]));
+
+          done();
+        });
+      });
+  }
+
+  private testSerializerAndGzipCalledOnce(done: () => void) {
+    const exporterClass = this.getExporter();
+    const exporter = new exporterClass(this.getEndpoint() + this.getEndpointPath(), {
+      compression: CompressionAlgorithm.GZIP,
+    });
+
+    const serializeStub = this.sandbox
+      .stub(exporter.parentSerializer, 'serializeRequest')
+      .returns(new Uint8Array([1, 2, 3]));
+    const gzipStub = this.sandbox.stub(require('zlib'), 'gzipSync').returns(new Uint8Array([0x1f, 0x8b, 1, 2, 3]));
+
+    exporter
+      .export([], (result: ExportResult) => {
+        expect(result.code).toBe(ExportResultCode.SUCCESS);
+        expect(serializeStub.callCount).toBe(1);
+        expect(gzipStub.callCount).toBe(1);
+        done();
+      })
+      .then(() => {
+        this.scope.on('request', (req, interceptor, body) => {
+          this.assertHeaders(req.headers);
+          expect(req.headers['content-encoding']).toBe('gzip');
+        });
+        done();
```

Comment:
`done()` should only be called once, and after everything is validated.
```suggestion
        expect(gzipStub.callCount).toBe(1);
      })
      .then(() => {
        this.scope.on('request', (req, interceptor, body) => {
          this.assertHeaders(req.headers);
          expect(req.headers['content-encoding']).toBe('gzip');
          done();
        });
```